avoids https://github.com/elm-lang/elm-compiler/issues/1591
--- result/Album.elm	1969-12-31 19:00:01.000000000 -0500
+++ Album.elm	2017-11-27 20:46:49.746198333 -0500
@@ -41,7 +41,7 @@
 jsonDecAlbumOrList : Json.Decode.Decoder ( AlbumOrList )
 jsonDecAlbumOrList =
     let jsonDecDictAlbumOrList = Dict.fromList
-            [ ("List", Json.Decode.map List (jsonDecAlbumList))
+            [ ("List", Json.Decode.map List (Json.Decode.lazy (\_ -> jsonDecAlbumList)))
             , ("Leaf", Json.Decode.map Leaf (jsonDecAlbum))
             ]
     in  decodeSumObjectWithSingleField  "AlbumOrList" jsonDecDictAlbumOrList
